src/utils/getUserRewards.ts

```diff
@@ -128,14 +274,48 @@ const computeBorrowIndex = async (market: Market, currentTimestamp: BigNumberish
     provider
   );
 
+const computeBorrowIndexes = async (market: Market, currentTimestamp: BigNumberish, provider: providers.Provider) => {
+  const rateType = "borrowRate";
+  const marketAddress = market.address;
+
+  const totalBorrowP2P = WadRayMath.wadMul(market.scaledBorrowInP2P, market.lastP2PBorrowIndex);
+  const totalBorrowOnPool = WadRayMath.wadMul(market.scaledBorrowOnPool, market.lastPoolBorrowIndex);
+  const totalBorrow = totalBorrowOnPool.add(totalBorrowP2P);
+  const lastPercentSpeed = totalBorrow.isZero()
+    ? constants.Zero
+    : totalBorrowP2P.mul(PercentMath.BASE_PERCENT).div(totalBorrow);
+  return {
+    p2pBorrowIndex: await computeIndex(
+      marketAddress,
+      market.p2pBorrowIndex,
+      market.borrowUpdateBlockTimestamp,
+      currentTimestamp,
+      rateType,
+      market.scaledBorrowInP2P,
+      provider,
+      (emission) => PercentMath.percentMul(emission, lastPercentSpeed)
+    ),
+    poolBorrowIndex: await computeIndex(
+      marketAddress,
+      market.poolBorrowIndex,
+      market.borrowUpdateBlockTimestamp,
+      currentTimestamp,
+      rateType,
+      market.scaledBorrowOnPool,
+      provider,
+      (emission) => PercentMath.percentMul(emission, PercentMath.BASE_PERCENT.sub(lastPercentSpeed))
```

Comment:
Just discovered a wonderful library last week you won't believe your eyes: https://github.com/Rubilmax/ethers-maths

Reply:
@julien-devatom fork that it's licensed MIT gogogo